src/zenml/integrations/kubeflow/metadata_stores/kubeflow_metadata_store.py

```diff
@@ -60,12 +88,195 @@ def get_tfx_metadata_config(
+
+    @property
+    def is_provisioned(self) -> bool:
+        """If the component provisioned resources to run locally."""
+        return fileio.exists(self.root_directory)
+
+    @property
+    def is_running(self) -> bool:
+        """If the component is running locally."""
+        if sys.platform != "win32":
+            if not check_if_daemon_is_running(self._pid_file_path):
+                return False
+        else:
+            pass
+        return True
+
+    def provision(self) -> None:
+        """Provisions resources to run the component locally."""
+        logger.info("Provisioning local Kubeflow Pipelines deployment...")
+        fileio.makedirs(self.root_directory)
+
+    def deprovision(self) -> None:
+        """Deprovisions all local resources of the component."""
+
+        if fileio.exists(self._log_file):
+            fileio.remove(self._log_file)
+
+        logger.info("Local kubeflow pipelines deployment deprovisioned.")
+
+    def resume(self) -> None:
+        """Resumes the local k3d cluster."""
+        if self.is_running:
+            logger.info("Local kubeflow pipelines deployment already running.")
+            return
+
+        self.start_kfp_metadata_daemon()
+        self.wait_until_metadata_store_ready()
+
+    def suspend(self) -> None:
+        """Suspends the local k3d cluster."""
+        if not self.is_running:
+            logger.info("Local kubeflow pipelines deployment not running.")
+            return
+
+        self.stop_kfp_metadata_daemon()
+
+    def start_kfp_metadata_daemon(self) -> None:
+        """Starts a daemon process that forwards ports so the Kubeflow Pipelines
+        Metadata MySQL database is accessible on the localhost."""
+
+        command = [
+            "kubectl",
+            "--context",
+            self.kubernetes_context,
+            "--namespace",
+            "kubeflow",
+            "port-forward",
+            "svc/metadata-grpc-service",
+            f"{self.port}:8080",
+        ]
+
+        if sys.platform == "win32":
+            logger.warning(
+                "Daemon functionality not supported on Windows. "
+                "In order to access the Kubeflow Pipelines Metadata locally, "
+                "please run '%s' in a separate command line shell.",
+                self.port,
+                " ".join(command),
+            )
+        elif not networking_utils.port_available(self.port):
+            logger.warning(
+                "Unable to port-forward Kubeflow Pipelines Metadata to local "
+                "port %d because the port is occupied. In order to access the "
+                "Kubeflow Pipelines Metadata locally, please change the metadata "
+                "store configuration to use a free port.",
+                self.port,
+            )
+        else:
+            from zenml.utils import daemon
+
+            def _daemon_function() -> None:
+                """Port-forwards the Kubeflow Pipelines Metadata pod."""
```

Comment:
```suggestion
                """Forwards the port of the Kubeflow Pipelines Metadata pod ."""
```
Not sure if it's the pod port that's being forwarded, or something else, but in any case thought it'd be clearer this way than using the 'port-forwards' verb which I hadn't seen before. If it's common, feel free to ignore me :)